src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs

```diff
@@ -42,253 +42,250 @@ internal static void AssertOnlyValidationMessage(this IEnumerable<ValidationMess
     [Collection("Validation Tests")]
     public partial class SwaggerModelerValidationTests
     {
-        private IEnumerable<ValidationMessage> ValidateSwagger(string input, ServiceDefinitionDocumentType serviceDefDocType = ServiceDefinitionDocumentType.ARM, 
-            ServiceDefinitionDocumentState mergeState = ServiceDefinitionDocumentState.Composed)
+        private IEnumerable<ValidationMessage> ValidateSwagger(string input, ServiceDefinitionMetadata metadata)
         {
-
             // Most rules are to be applied for ARM documents
             // Also, most rules need to be run over the composite document (i.e. AFTER merge state)
             // hence the defaults
             using (NewContext)
             {
                 var validator = new RecursiveObjectValidator(PropertyNameResolver.JsonName);
                 var serviceDefinition = SwaggerParser.Parse(input, File.ReadAllText(input));
-
-                var metaData = new ServiceDefinitionMetadata
-                {
-                    ServiceDefinitionDocumentType = serviceDefDocType,
-                    MergeState = mergeState
-                };
-                
-                return validator.GetValidationExceptions(new Uri(input, UriKind.RelativeOrAbsolute), serviceDefinition, metaData).OfType<ValidationMessage>();
+                return validator.GetValidationExceptions(new Uri(input, UriKind.RelativeOrAbsolute), serviceDefinition, metadata).OfType<ValidationMessage>();
             }
         }
 
+        private ServiceDefinitionMetadata GetMetadataForRuleTest(Type rType)
+        {
+            var rule = (Rule)Activator.CreateInstance(rType);
+            return new ServiceDefinitionMetadata
+            {
+                ServiceDefinitionDocumentType = rule.ServiceDefinitionDocumentType,
+                MergeState = rule.ValidationRuleMergeState
+            };
+        }
+
         [Fact]
         public void MissingDescriptionValidation()
         {
```

Comment:
given the structure of all these tests, and the fact that we're now repeating the *same type name twice* I would refactor further. It's even the same directory structure, so how about
``` C#
private IEnumerable<ValidationMessage> GetMessagesForRule<TRule>(string swaggerFileName)
{
    var messages = ValidateSwagger(Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "Validation", swaggerFileName), GetMetadataForRuleTest<TRule>());
    return messages.Where(message => message.Rule is TRule);
}
```

so that the tests look like
``` c#
var messages = GetMessagesForRule<DescriptionAndTitleMissing>("definition-missing-description.json");
Assert.Equal(2, messages.Count());
```
With this structure, the `AssertOnlyValidationMessage` functions can also be simplified (and partially removed) as they don't filter the message type anymore. Honestly, I'd just make a helper method `JustWarnings` such that you can do
``` c#
Assert.Equal(2, messages.JustWarnings().Count());
```
in places where we had `AssertOnlyValidationWarning`.